factory/builder.py

```diff
@@ -245,6 +246,13 @@ def build(self, parent_step=None, force_sequence=None):
             base_post=self.factory_meta.post_declarations,
         )
 
+        # TODO: come up with a better solution
```

Comment:
I think in this case we should leave a comment on why this code branch exists and leave it as is. It's not the cleanest but it works.

Reply:
I was actually asking for a _better_ solution. I don't like it, even though it "works" :/

Reply:
@francoisfreitag do you have any ideas on how to improve this?